pennylane/optimize/lie_algebra.py

```diff
@@ -288,7 +288,7 @@ def step_and_cost(self):
             lie_gradient, self.stepsize, self.trottersteps, self.exact
         )(self.circuit.func)
 
-        self.circuit = qml.QNode(new_circuit, self.circuit.device)
+        self.circuit = qml.QNode(new_circuit, self.circuit.device, diff_method=None)
```

Comment:
How come passing `diff_method=None` was required?

Reply:
It was required because `diff_method="best"` defaults to parameter-shift, which if I'm not mistaken calls the interface-specific `execute` function which we don't support yet.

The gradient of the QNode is computed directly in the optimizer, so we don't need the QNode itself to be differentiable anyways.

Reply:
Thanks! Worth making a small comment here about
> The gradient of the QNode is computed directly in the optimizer, so we don't need the QNode itself to be differentiable anyways.